src/libraries/Microsoft.Extensions.Hosting/src/ValidatorOptions.cs

```diff
@@ -0,0 +1,14 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.Extensions.DependencyInjection
+{
+    internal class ValidatorOptions
+    {
+        // Maps each options type to a method that forces its evaluation, e.g. IOptionsMonitor<TOptions>.Get(name)
+        public IDictionary<Type, Action> Validators { get; } = new Dictionary<Type, Action>();
```

Comment:
I know this is an internal implementation detail, but is there any particular reason to use a dictionary over a list of actions for this?

Reply:
True

Reply:
My understanding is that for a `TOptions`, we only need to run the `Action` once. If `ValidateOnStart` is called multiple times for the same `TOptions`, we only want it to be "validated" once.

Reply:
Ah yea thats why I did it 😄

Reply:
@maryamariyan - it probably isn't worth resetting CI ... but since I had the same question initially, maybe add a comment here explaining why a Dictionary is used next time you are touching this code.